extide/gradle/src/org/netbeans/modules/gradle/spi/execute/JavaRuntimeManager.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.netbeans.modules.gradle.spi.execute;
+
+import java.io.File;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+import javax.swing.event.ChangeListener;
+
+/**
+ *
+ * @author Laszlo Kishalmi
+ */
+public interface JavaRuntimeManager {
+
+    public static String DEFAULT_RUNTIME_ID = "default_platform"; //NOI18N
+    
+    Set<JavaRuntime> getAvailableRuntimes();
+
+    default Optional<Runnable> manageRuntimesAction() {
```

Comment:
can the caller of this method use the explicit information that there is no Runnable available?
if not then maybe just return () -> {} (empty runnable) and let the caller be ignorant

Reply:
Yes it is actually used that way in the Customizer.